src/Elements/utilities.jl

```diff
@@ -0,0 +1,35 @@
+["Helper functions."]
+
+"""
+Find the final configuration fo node `n`.
+
+Search begins with the node `n` itself followed by any parent `Node` configuration setting.
+"""
+function findconfig(n::Node)
+    local config
+    # Stage 1: nodes's config.
+    config = n.data
+
+    # Stage 2: Parent's config.
+    while isdefined(n, :parent)
+        n = n.parent
+        update!(config, n.data)
+    end
+    config
+end
```

Comment:
``findconfig`` shouldn't be mutating any of the nodes' ``config``s. Just search for the closest config and return that. Or even better, require a ``Symbol`` arg to ``findconfig(n::Node, field::Symbol)`` and just return the field's value.

Basically, once created the ``document(...)`` object shouldn't be changed, just used for lookup otherwise it's going to get difficult to debug if it's getting changed internally.